hudi-common/src/main/java/org/apache/hudi/common/model/HoodieSimpleMergeKey.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.hudi.common.model;
+
+import java.io.Serializable;
+import java.util.Objects;
+
+/**
+ * Wraps {@link HoodieKey} and implements the {@link HoodieMergeKey} interface for simple scenarios where the key is a string.
+ */
+public class HoodieSimpleMergeKey implements HoodieMergeKey {
```

Comment:
Actually we might could not call this a "simple" merge key, becase the it looks like the recordk key here can take multiple fields in table schema.

Reply:
record key fields can be multiple but it's still a string combined.

Reply:
I'm talking about the notion instead of the physical impl.

Reply:
Ok, would renaming to `HoodiePrimaryMergeKey` be more suitable, more like how database have primary and composite key? The basic idea is that this merge key is based on record key field(s) which will be unique.

Reply:
This new api is no longer needed.